README.rst
Python Client for Dialogflow API
================================

|stable| |pypi| |versions|

`Dialogflow API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-dialogflow.svg
   :target: https://pypi.org/project/google-cloud-dialogflow/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-dialogflow.svg
   :target: https://pypi.org/project/google-cloud-dialogflow/
.. _Dialogflow API: https://www.dialogflow.com/
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/dialogflow/latest
.. _Product Documentation:  https://www.dialogflow.com/

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Dialogflow API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Dialogflow API.:  https://www.dialogflow.com/
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-dialogflow


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-dialogflow

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Dialogflow API
   to see other available methods on the client.
-  Read the `Dialogflow API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Dialogflow API Product documentation:  https://www.dialogflow.com/
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

[PyPI History][1]

[1]: https://pypi.org/project/dialogflow/#history

## [2.19.2](https://github.com/googleapis/python-dialogflow/compare/v2.19.1...v2.19.2) (2023-01-24)


### Documentation

* Reformat some comments and documentation ([#614](https://github.com/googleapis/python-dialogflow/issues/614)) ([8123d1b](https://github.com/googleapis/python-dialogflow/commit/8123d1b5c4ffea9d7c6da1958a8e662410073d04))

## [2.19.1](https://github.com/googleapis/python-dialogflow/compare/v2.19.0...v2.19.1) (2023-01-20)


### Bug Fixes

* Add context manager return types ([c5272ff](https://github.com/googleapis/python-dialogflow/commit/c5272ffd954eb42e6ed78cb17300737c36d1e9b1))


### Documentation

* Add documentation for enums ([c5272ff](https://github.com/googleapis/python-dialogflow/commit/c5272ffd954eb42e6ed78cb17300737c36d1e9b1))

## [2.19.0](https://github.com/googleapis/python-dialogflow/compare/v2.18.0...v2.19.0) (2023-01-14)


### Features

* **v2:** Added SuggestConversationSummary RPC ([#607](https://github.com/googleapis/python-dialogflow/issues/607)) ([e5aabc8](https://github.com/googleapis/python-dialogflow/commit/e5aabc8c2022b943b793c20d9f2280c9db24001c))

## [2.18.0](https://github.com/googleapis/python-dialogflow/compare/v2.17.0...v2.18.0) (2023-01-10)


### Features

* Add support for python 3.11 ([#604](https://github.com/googleapis/python-dialogflow/issues/604)) ([08a6580](https://github.com/googleapis/python-dialogflow/commit/08a6580b7bb5ba7e9efab0ad5d2bc141bde45572))

## [2.17.0](https://github.com/googleapis/python-dialogflow/compare/v2.16.1...v2.17.0) (2022-12-15)


### Features

* Add support for `google.cloud.dialogflow.__version__` ([234ab9b](https://github.com/googleapis/python-dialogflow/commit/234ab9be67c4d8b3853dc20f38ec505cfeabb8fe))
* Add typing to proto.Message based class attributes ([234ab9b](https://github.com/googleapis/python-dialogflow/commit/234ab9be67c4d8b3853dc20f38ec505cfeabb8fe))
* Added cx_current_page field to AutomatedAgentReply  ([234ab9b](https://github.com/googleapis/python-dialogflow/commit/234ab9be67c4d8b3853dc20f38ec505cfeabb8fe))
* **v2:** Added obfuscated_external_user_id to Participant ([f8c3d35](https://github.com/googleapis/python-dialogflow/commit/f8c3d35eedcab6996b18ba08c0b4637699d5e0e3))
* **v2:** Added StreamingAnalyzeContent API ([#590](https://github.com/googleapis/python-dialogflow/issues/590)) ([f8c3d35](https://github.com/googleapis/python-dialogflow/commit/f8c3d35eedcab6996b18ba08c0b4637699d5e0e3))
* **v2beta1:** Add ability to set Cloud Speech model in SpeechToTextConfig ([#587](https://github.com/googleapis/python-dialogflow/issues/587)) ([1d258c2](https://github.com/googleapis/python-dialogflow/commit/1d258c2ea8e9993a382dfb318f3022739030ad04))
* **v2:** Can directly set Cloud Speech model on the SpeechToTextConfig ([f8c3d35](https://github.com/googleapis/python-dialogflow/commit/f8c3d35eedcab6996b18ba08c0b4637699d5e0e3))


### Bug Fixes

* Add dict typing for client_options ([234ab9b](https://github.com/googleapis/python-dialogflow/commit/234ab9be67c4d8b3853dc20f38ec505cfeabb8fe))
* **deps:** Require google-api-core &gt;=1.34.0, >=2.11.0 ([234ab9b](https://github.com/googleapis/python-dialogflow/commit/234ab9be67c4d8b3853dc20f38ec505cfeabb8fe))
* Drop usage of pkg_resources ([234ab9b](https://github.com/googleapis/python-dialogflow/commit/234ab9be67c4d8b3853dc20f38ec505cfeabb8fe))
* Fix timeout default values ([234ab9b](https://github.com/googleapis/python-dialogflow/commit/234ab9be67c4d8b3853dc20f38ec505cfeabb8fe))


### Documentation

* Clarified docs for Sentiment ([234ab9b](https://github.com/googleapis/python-dialogflow/commit/234ab9be67c4d8b3853dc20f38ec505cfeabb8fe))
* **samples:** Snippetgen handling of repeated enum field ([234ab9b](https://github.com/googleapis/python-dialogflow/commit/234ab9be67c4d8b3853dc20f38ec505cfeabb8fe))
* **samples:** Snippetgen should call await on the operation coroutine before calling result ([234ab9b](https://github.com/googleapis/python-dialogflow/commit/234ab9be67c4d8b3853dc20f38ec505cfeabb8fe))

## [2.16.1](https://github.com/googleapis/python-dialogflow/compare/v2.16.0...v2.16.1) (2022-10-10)


### Bug Fixes

* **deps:** Allow protobuf 3.19.5 ([#585](https://github.com/googleapis/python-dialogflow/issues/585)) ([4030787](https://github.com/googleapis/python-dialogflow/commit/40307877bbc467e6475a5614d315de4e6d902683))


### Documentation

* **samples:** Add sample code for StreamingAnalyzeContent for live transcription. ([#575](https://github.com/googleapis/python-dialogflow/issues/575)) ([b6bbd62](https://github.com/googleapis/python-dialogflow/commit/b6bbd62a3d3293e3ae031b2fdd8940343da08e74))

## [2.16.0](https://github.com/googleapis/python-dialogflow/compare/v2.15.2...v2.16.0) (2022-10-03)


### Features

* Include conversation dataset name to be created with dataset creation metadata ([#582](https://github.com/googleapis/python-dialogflow/issues/582)) ([74d45fb](https://github.com/googleapis/python-dialogflow/commit/74d45fb4c69d6a150aa86bf249ba291e6bd72e13))
* **v2beta1:** Add Agent Assist Summarization API (https://cloud.google.com/agent-assist/docs/summarization) ([1648b27](https://github.com/googleapis/python-dialogflow/commit/1648b27b87416806ea7fcc016ad3caa5f89f5789))


### Bug Fixes

* **deps:** Require protobuf >= 3.20.2 ([#579](https://github.com/googleapis/python-dialogflow/issues/579)) ([2159259](https://github.com/googleapis/python-dialogflow/commit/21592593540d1e058be7debf565f7b737a45f79b))


### Documentation

* clarify SuggestionFeature enums which are specific to chat agents ([74d45fb](https://github.com/googleapis/python-dialogflow/commit/74d45fb4c69d6a150aa86bf249ba291e6bd72e13))
* **v2beta1:** clarify SuggestionFeature enums which are specific to chat agents ([1648b27](https://github.com/googleapis/python-dialogflow/commit/1648b27b87416806ea7fcc016ad3caa5f89f5789))


noxfile.py
# Copyright 2017, Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from __future__ import absolute_import
import os
import shutil

import nox

BLACK_VERSION = "black==22.3.0"
BLACK_PATHS = [
    "dialogflow",
    "dialogflow_v2",
    "dialogflow_v2beta1",
    "tests",
    "docs",
    "noxfile.py",
    "setup.py",
]
DEFAULT_PYTHON_VERSION = "3.8"

# Error if a python version is missing
nox.options.error_on_missing_interpreters = True


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint(session):
    """Run linters.
    Returns a failure if the linters find linting errors or sufficiently
    serious code quality issues.
    """
    session.install("flake8", BLACK_VERSION)
    session.run("black", "--check", *BLACK_PATHS)
    session.run("flake8", "dialogflow", "dialogflow_v2", "dialogflow_v2beta1")


@nox.session(python=DEFAULT_PYTHON_VERSION)
def blacken(session):
    """Run black.
    Format code to uniform standard.
    This currently uses Python 3.6 due to the automated Kokoro run of synthtool.
    That run uses an image that doesn't have 3.6 installed. Before updating this
    check the state of the `gcp_ubuntu_config` we use for that Kokoro run.
    """
    session.install(BLACK_VERSION)
    session.run("black", *BLACK_PATHS)


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint_setup_py(session):
    """Verify that setup.py is valid (including RST check)."""
    session.install("docutils", "pygments")
    session.run("python", "setup.py", "check", "--restructuredtext", "--strict")


def default(session):
    # Install all test dependencies, then install this package in-place.
    session.install("mock", "pytest", "pytest-cov")
    session.install("-e", ".")

    # Run py.test against the unit tests.
    session.run(
        "py.test",
        "--quiet",
        "--cov=dialogflow",
        "--cov=dialogflow_v2beta1",
        "--cov=dialogflow_v2" "--cov-append",
        "--cov-config=.coveragerc",
        "--cov-report=",
        "--cov-fail-under=0",
        os.path.join("tests", "unit"),
        *session.posargs,
    )


@nox.session(python=["2.7", "3.5", "3.6", "3.7", "3.8"])
def unit(session):
    """Run the unit test suite."""
    default(session)


@nox.session(python=DEFAULT_PYTHON_VERSION)
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=96")

    session.run("coverage", "erase")


@nox.session(python=DEFAULT_PYTHON_VERSION)
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "jinja2<3.1", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
